lasagne/layers/helper.py

```diff
@@ -148,7 +148,15 @@ def get_output(layer_or_layers, inputs=None, **kwargs):
     both, while the latter will use two different dropout masks.
     """
     from .input import InputLayer
-    from .base import MergeLayer
+    from .base import MergeLayer, Layer
+    # check if the keys of the dictionary are valid
+    if isinstance(inputs, dict):
+        for input_key in inputs.keys():
+            if not isinstance(input_key, Layer):
+                if input_key:
```

Comment:
This is too weak, this would also allow `False` or `''` or `0` as a key. Tests for `None` should better be explicit. I'd suggest the following (it does some things slightly differently):
```python
# check if the keys of the dictionary are valid
if isinstance(inputs, dict):
    for key in inputs:
        if (key is not None) and (not isinstance(key, Layer)):
            raise TypeError(... % type(key))
```